src/main/cpp/src/cast_string_to_datetime.cu

```diff
@@ -697,8 +705,11 @@ __device__ result_type parse_timestamp_string(unsigned char const* const ptr,
 /**
  * Parse a string with timezone to a timestamp.
  */
-template <bool is_spark_320>
 struct parse_timestamp_string_fn {
+  // version info
+  bool is_spark_320;
```

Comment:
NIT: IMHO, it is better to use template variable to dispatch versions to avoid branch operation being executed in each CUDA thread.  Perhaps, we can handle the version dispatch more elegantly just like type dispatcher of CUDF (using template functions).

Of course, current implementation looks good enough for me. Maybe we can mark it as a TODO action?